public/app/features/variables/state/sharedReducer.ts

```diff
@@ -16,13 +17,15 @@ const sharedReducerSlice = createSlice({
   reducers: {
     addVariable: (state: VariablesState, action: PayloadAction<VariablePayload<AddVariable>>) => {
       const id = action.payload.id ?? action.payload.data.model.name; // for testing purposes we can call this with an id
+      const initialState = cloneDeep(variableAdapters.get(action.payload.type).initialState);
+
       const variable = {
-        ...cloneDeep(variableAdapters.get(action.payload.type).initialState),
-        ...action.payload.data.model,
+        ...lodashDefaults(action.payload.data.model, initialState),
```

Comment:
Marcus and I looked at the tests and action.payload.data.model is mutated here by lodash/defaults. Marcus will push changes